src/explorer/CosmosDBConnection.ts

```diff
@@ -29,6 +30,32 @@ export class CosmosDBConnection extends AzureTreeItem<ISiteTreeRoot> {
     }
 
     public async deleteTreeItemImpl(): Promise<void> {
-        throw new Error('Not implemented yet!');
+        const valueToDelete = this.cosmosDBDatabase.connectionString;
+
+        const appSettings = await this.root.client.listApplicationSettings();
+        const properties = appSettings.properties;
+        if (properties) {
+            const keysToDelete: string[] = [];
+            Object.keys(properties).forEach((key) => {
+                if (properties[key] === valueToDelete) {
+                    keysToDelete.push(key);
+                }
+            });
+
+            if (keysToDelete.length > 0) {
+                const warning: string = `Are you sure you want to remove connection "${this.label}"? This will delete the following application settings: ${keysToDelete.map((s) => `"${s}"`).join(', ')}.`;
+                const items: vscode.MessageItem[] = [DialogResponses.deleteResponse, DialogResponses.cancel];
+                const result: vscode.MessageItem = await ext.ui.showWarningMessage(warning, { modal: true }, ...items);
+                if (result === DialogResponses.cancel) {
+                    throw new UserCancelledError();
+                }
+            }
+            keysToDelete.forEach((key) => {
+                delete properties[key];
+            });
+            appSettings.properties = properties;
```

Comment:
nit: I don't think you have to do `appSettings.properties = properties;` since I think `properties` is a reference (aka not a copy)